src/compiler/checker.ts

```diff
@@ -27872,6 +27872,12 @@ namespace ts {
                 return resolveUntypedCall(node);
             }
 
+            if (isArrayLiteralExpression(node.parent)) {

```

Comment:
I would move this into the `!callSignatures.length` check because there is technically a way to make this work in the type system.

```ts
interface String {
    (...args: any[]): any;
}

// should not error.
let x = [
  `hello`
  `world`
];
```

Reply:
Done